File: Data Structures & Algorithms/duplicate-integer/submission-2.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        var visit = new HashSet<int>();
        for(int i=0;i<nums.Count();i++){
            if(visit.Contains(nums[i])) return true;
            visit.Add(nums[i]);
        }
        return false;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<List<Integer>> res = new HashSet();
        Arrays.sort(nums);
        for(int i=0;i<nums.length-2;i++){
            int start = i+1;
            int end = nums.length - 1;
            while(start<end){
                if(nums[i]+nums[start]+nums[end] == 0){
                    List<Integer> sol = new ArrayList();
                    sol.add(nums[i]);
                    sol.add(nums[start]);
                    sol.add(nums[end]);
                    res.add(sol);
                    start++;
                }
                if(nums[i]+nums[start]+nums[end] < 0) start++;
                if(nums[i]+nums[start]+nums[end] > 0) end--;
            }
        }
        return new ArrayList(res);
    }
}
